### ISSUE TITLE
Support 'version.digest' in resource_type of type 'registry-image'

### ISSUE THREAD

**Issue by Kitanotori** [concourse/concourse#8512]
### Discussed in https://github.com/concourse/concourse/discussions/6166

<div type='discussions-op-text'>

<sup>Originally posted by **Krenair** October 17, 2020</sup>
I'd like to be able to ensure that a resource type uses a specific image, compared against a sha256 digest - to ensure that no one with permissions on the Docker registry has tampered with the image.

The registry-image docs at https://github.com/concourse/registry-image-resource#use-in-tasks talk about a `version` struct outside of `source`.
However, https://concourse-ci.org/resource-types.html#schema.resource_type only talks about passing the `source` through to registry-image, nothing about `version`.</div>

**Comment by Kitanotori** [comment]
The lack of 'digest' support for resource type definition might even be considered a bug, as the lack of this feature breaks predictability of builds - image tags are just pointers that can change. Also, I believe that by using digest we can avoid unnecessary registry calls for checking latest image version pointed by a tag.

**Comment by Kitanotori** [comment]
@marco-m-pix4d I see, thank you for clarification.

**Comment by Kitanotori** [comment]
@marco-m-pix4d Just confirmation; does [registry-image-resource](https://github.com/concourse/registry-image-resource) also apply to `resource_types` property? I want to specify the image digest of my custom resource via `resource_types[].version.digest`, not via `resources[].version.digest`.

**Comment by marco-m-pix4d** [comment]
> Just confirmation; does [registry-image-resource](https://github.com/concourse/registry-image-resource) also apply to resource_types property?

Yes. Actually it only applies to `resource_types`, not to `resources`. It takes a while to get one's brain to understand the differences and see clearly, everybody passed through this :-)
